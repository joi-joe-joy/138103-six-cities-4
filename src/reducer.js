import {extend} from "./utils.js";
import offers from "./mocks/offers.js";

let initCity = offers[0].city;
let initOffers = offers.filter((offer) => offer.city === initCity);

// Получаем все значения городов
let cities = offers.map((offer)=>offer.city);
// Получаем уникальные значения городов
let citiesList = Array.from(new Set(cities));

const initialState = {
  city: initCity,
  offersCityList: initOffers,
  offers,
  citiesList
};

const ActionType = {
  CHANGE_CITY: `CHANGE_CITY`,
  GET_OFFERS_CITY_LIST: `GET_OFFERS_CITY_LIST`,
};

const ActionCreator = {
  changeCity: (city) => {
    let changedCity = city ? city : offers[0].city;

    return {
      type: ActionType.CHANGE_CITY,
      payload: changedCity
    };
  },
  getOffersCityList: (city) => {
    let offersCityList = offers.filter((offer) => offer.city === city);

    return {
      type: ActionType.GET_OFFERS_CITY_LIST,
      payload: offersCityList
    };
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_CITY:
      return extend(state, {
        city: action.payload
      });
    case ActionType.GET_OFFERS_CITY_LIST:
      return extend(state, {
        offersCityList: action.payload
      });
  }

  return state;
};

export {reducer, ActionType, ActionCreator};
