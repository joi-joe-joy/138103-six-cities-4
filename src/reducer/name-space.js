export const NameSpace = {
  DATA: `DATA`,
  USER: `USER`,
  PLACE: `PLACE`
};
