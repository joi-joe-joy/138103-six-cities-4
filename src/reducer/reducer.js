import {combineReducers} from "redux";
import {reducer as data} from "./data/data";
import {reducer as place} from "./place/place";
import {reducer as user} from "./user/user";
import {reducer as comments} from "./comments/comments";
import {reducer as favorite} from "./favorite/favorite";
import {NameSpace} from "./name-space";

export default combineReducers({
  [NameSpace.DATA]: data,
  [NameSpace.PLACE]: place,
  [NameSpace.USER]: user,
  [NameSpace.COMMENTS]: comments,
  [NameSpace.FAVORITE]: favorite,
});
