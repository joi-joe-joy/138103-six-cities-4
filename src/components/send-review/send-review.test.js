import * as React from "react";
import renderer from "react-test-renderer";
import SendReview from "./send-review";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {NameSpace} from "../../reducer/name-space";

const mockStore = configureStore([]);

const children = <div/>;

it(`Renderer SendReview correctly`, () => {
  const store = mockStore({
    [NameSpace.COMMENTS]: {
      comments: {}
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <SendReview
          onSubmit={()=>{}}
          rating={5}
          comment={``}
        >
          {children}
        </SendReview>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
