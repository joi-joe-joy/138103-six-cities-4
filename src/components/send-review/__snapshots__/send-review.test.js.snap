// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer SendReview correctly 1`] = `
<form
  action="#"
  className="reviews__form form"
  method="post"
  onSubmit={[Function]}
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
        50 characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
